File: Java/Array/TripletSum.java
import java.util.Arrays;

public class TripletSum {
    public static boolean hasTriplet(int[] arr, int target){
        Arrays.sort(arr);
        int n = arr.length;

        for(int i=0; i<n-2; i++){
            int l = i + 1, r = n - 1;
            int res = target - arr[i];

            while(l < r){
                if(arr[l] + arr[r] == res){
                    return true;
                }

                if(arr[l] + arr[r] < res) l++;
                else if(arr[l] + arr[r] > res) r--;
            }
        }
        return false;
    }
    public static void main(String[] args) {
        int[] arr = {1, 4, 45, 6, 10, 8};
        int target = 13;
        System.out.println(TripletSum.hasTriplet(arr, target));
    }
}

File: Golang/Array/TripletSum.go
package main

import (
	"sort"
	"fmt"
)

func hasTriplet(arr []int, target int) bool {
	n := len(arr)
	sort.Ints(arr)

	for i:=0; i<n-2; i++{
		l, r := i + 1, n - 1
		res := target - arr[i]

		for l < r{
			sum := arr[l] + arr[r]

			if sum == res{
				return true
			}
			if sum > res{
				r--
			}else{
				l++
			}
		}
	}
	return false
}

func main(){
	arr := []int{1, 4, 45, 6, 10, 8}
	fmt.Println(hasTriplet(arr, 13))
}
